python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -483,9 +520,30 @@ def debug(self):
             logger.error(result.get('Message', ''))
 
     @staticmethod
-    def _getStateTime(job, status):
+    def _bulk_updateStateTime( jobStateDict ):
+        """ This performs the same as the _getStateTime method but loops over a list of job ids within the DIRAC namespace (much faster)
+        Args:
+            jobStateDict (dict): This is a dict of {job.backend.id : job_status, } elements
+        """
+        for this_state in jobStateDict:
+            bulk_result = execute("getBulkStateTime(\'%s\',\'%s\')" %
+                                    (repr([job.backend.id for job in jobStateDict[this_state]]), this_state))
+            for this_job in jobStateDict[this_state]:
+                if bulk_result[this_job.backend.id]:
+                    DiracBase._getStateTime(this_job, this_state, bulk_result[this_job.backend.id])
+                else:
+                    DiracBase._getStateTime(this_job, this_state)
```

Comment:
Is this `if`/`else` needed? Doesn't `bulk_result[this_job.backend.id]` default to `{}` anyway?